swarm_copy/app/database/db_utils.py

```diff
@@ -10,39 +10,65 @@
 from swarm_copy.app.database.sql_schemas import Messages, Threads, utc_now
 
 
+async def get_thread(user_id: str, thread_id: str, session: AsyncSession) -> Threads:
+    """Check if the current thread / user matches."""
+    thread_result = await session.execute(
+        select(Threads).where(
+            Threads.user_id == user_id, Threads.thread_id == thread_id
+        )
+    )
+    thread = thread_result.scalars().one_or_none()
+    if not thread:
+        raise HTTPException(
+            status_code=404,
+            detail={
+                "detail": "Thread not found.",
+            },
+        )
+    return thread
+
+
 async def put_messages_in_db(
-    history: list[dict[str, Any]], thread_id: str, offset: int, session: AsyncSession
+    history: list[dict[str, Any]],
+    user_id: str,
+    thread_id: str,
+    offset: int,
+    session: AsyncSession,
 ) -> None:
     """Add the new messages in the database."""
     for i, messages in enumerate(history):
+        if messages["role"] != "assistant":
+            entity = str(messages["role"]).upper()
+        elif messages["content"]:
+            entity = "AI_MESSAGE"
+        else:
+            entity = "AI_TOOL"
+
         new_msg = Messages(
             order=i + offset,
             thread_id=thread_id,
+            entity=entity,
             content=json.dumps(messages),
         )
         session.add(new_msg)
+
     # we need to update the thread update time
-    thread = await session.get(Threads, thread_id)
-    if thread:
-        thread.update_date = utc_now()
-    else:
-        raise HTTPException(
-            status_code=404,
-            detail={
-                "detail": "Thread not found.",
-            },
-        )
+    thread = await get_thread(user_id=user_id, thread_id=thread_id, session=session)
+    thread.update_date = utc_now()
     await session.commit()
 
 
 async def get_messages_from_db(
-    thread_id: str, session: AsyncSession
+    user_id: str, thread_id: str, session: AsyncSession
```

Comment:
But how do we `get_thread()` without a `thread_id` ? 
I am not sure but a dependency takes no argument ?

Reply:
I could, it would only make sense for `qa/chat` and `qa/streamed`. For the threads we want different infos from the DB.
I could make a general `get_messages` dependency and make it only return the `Messages` object from the DB, it would be a one liner dependency though